Cannot store / serialize auto-generated Remote type

In order to make a more type-safe API, I was trying to store a typed address like:

```rust
#[cw_serde]
pub struct Config {
    pub adjustment: Decimal,

    /// Address of the contract we query for the price feed to normalize the foreign asset into native tokens.
    pub price_feed: price_feed_api::Remote,
}
```

However, the auto-generated Remote doesn't support most of those traits (even while Addr does). Maybe simply wrapping it in #[cw_serde] or doing something like Addr would be enough

```
error[E0277]: the trait bound `mesh_apis::price_feed_api::Remote<'_>: Clone` is not satisfied
  --> contracts/consumer/converter/src/state.rs:15:5
   |
5  | #[cw_serde]
   | ----------- in this procedural macro expansion
...
15 |     pub price_feed: price_feed_api::Remote,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `mesh_apis::price_feed_api::Remote<'_>`
   |
   = note: this error originates in the derive macro `::std::clone::Clone` which comes from the expansion of the attribute macro `cw_serde` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `mesh_apis::price_feed_api::Remote<'_>` doesn't implement `Debug`
  --> contracts/consumer/converter/src/state.rs:15:5
   |
5  | #[cw_serde]
   | ----------- in this procedural macro expansion
...
15 |     pub price_feed: price_feed_api::Remote,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `mesh_apis::price_feed_api::Remote<'_>` cannot be formatted using `{:?}` because it doesn't implement `Debug`
   |
   = help: the trait `Debug` is not implemented for `mesh_apis::price_feed_api::Remote<'_>`
   = note: this error originates in the derive macro `::std::fmt::Debug` which comes from the expansion of the attribute macro `cw_serde` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0369]: binary operation `==` cannot be applied to type `mesh_apis::price_feed_api::Remote<'_>`
  --> contracts/consumer/converter/src/state.rs:15:5
   |
5  | #[cw_serde]
   | ----------- in this procedural macro expansion
...
15 |     pub price_feed: price_feed_api::Remote,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `::std::cmp::PartialEq` which comes from the expansion of the attribute macro `cw_serde` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `mesh_apis::price_feed_api::Remote<'_>: JsonSchema` is not satisfied
   --> contracts/consumer/converter/src/state.rs:15:21
    |
15  |     pub price_feed: price_feed_api::Remote,
    |                     ^^^^^^^^^^^^^^^^^^^^^^ the trait `JsonSchema` is not implemented for `mesh_apis::price_feed_api::Remote<'_>`
    |
    = help: the following other types implement trait `JsonSchema`:
              &'a T
              &'a mut T
              ()
              (T0, T1)
              (T0, T1, T2)
              (T0, T1, T2, T3)
              (T0, T1, T2, T3, T4)
              (T0, T1, T2, T3, T4, T5)
            and 246 others
note: required by a bound in `SchemaGenerator::subschema_for`
   --> /Users/ethanfrey/.cargo/registry/src/github.com-1ecc6299db9ec823/schemars-0.8.12/src/gen.rs:218:38
    |
218 |     pub fn subschema_for<T: ?Sized + JsonSchema>(&mut self) -> Schema {
    |                                      ^^^^^^^^^^ required by this bound in `SchemaGenerator::subschema_for`

error[E0277]: the trait bound `mesh_apis::price_feed_api::Remote<'_>: JsonSchema` is not satisfied
 --> contracts/consumer/converter/src/state.rs:5:1
  |
5 | #[cw_serde]
  | ^^^^^^^^^^^ the trait `JsonSchema` is not implemented for `mesh_apis::price_feed_api::Remote<'_>`
```

--- a/sylvia-derive/src/remote.rs
+++ b/sylvia-derive/src/remote.rs
@@ -48,7 +48,7 @@ impl Remote {
         });
 
         quote! {
-            #[derive(#sylvia ::serde::Serialize, #sylvia ::serde::Deserialize)]
+            #[derive(#sylvia ::serde::Serialize, #sylvia ::serde::Deserialize, Clone, Debug, PartialEq, #sylvia ::schemars::JsonSchema)]
             pub struct Remote<'a>(std::borrow::Cow<'a, #sylvia ::cw_std::Addr>);
 
             impl Remote<'static> {

--- a/sylvia/tests/remote.rs
+++ b/sylvia/tests/remote.rs
@@ -1,3 +1,4 @@
+use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{Response, StdResult};
 use sylvia::contract;
 use sylvia::types::InstantiateCtx;
@@ -30,6 +31,13 @@ impl SomeContract {
     pub fn instantiate(&self, _ctx: InstantiateCtx) -> StdResult<Response> {
         Ok(Response::new())
     }
+}
+
+// Making sure `Remote` can be stored in `#[cw_serde]` types
+#[cw_serde]
+#[allow(dead_code)]
+struct CustomStorage {
+    remote: crate::Remote<'static>,
 }
 
 #[cfg(test)]

